care/emr/reports/discharge_summary.py

```diff
@@ -1,262 +1,191 @@
-        "files": files,
-        "care_team": care_team_display,
-        "discharge_summary_advice": encounter.discharge_summary_advice,
-    }
-
-
-def compile_typ(output_file, data):
-    try:
-        logo_path = (
-            Path(settings.BASE_DIR)
-            / "staticfiles"
-            / "images"
-            / "logos"
-            / "black-logo.svg"
-        )
+        section = section_cls(section_conf, ctx, renderer)
+        fragments.append(section.render())
 
-        data["logo_path"] = "black-logo.svg"
-        with tempfile.TemporaryDirectory() as tmpdir:
-            template = Path(tmpdir) / "template.typ"
-            template.write_text(
-                render_to_string(
-                    "reports/patient_discharge_summary_pdf_template.typ", context=data
-                )
-            )
+    extract_images(included_images, config)
 
-            logo_dest = Path(tmpdir) / "black-logo.svg"
-            logo_dest.write_text(logo_path.read_text())
-
-            subprocess.run(  # noqa: S603
-                [
-                    settings.TYPST_BIN,
-                    "compile",
-                    str(template.name),
-                    str(output_file),
-                ],
-                capture_output=True,
-                check=True,
-                shell=False,
-                cwd=tmpdir,
-            )
+    final_content = "\n\n".join([page_layout, header_content, *fragments])
+    return final_content, included_images
 
-        logging.info(
-            "Successfully Compiled Summary pdf for %s", data["encounter"].external_id
-        )
 
-    except subprocess.CalledProcessError as e:
-        logging.error(
-            "Error compiling summary pdf for %s: %s",
-            data["encounter"].external_id,
-            e.stderr.decode("utf-8"),
-        )
-        raise e
+def compile_typ(
+    output_file: str, template_code: str, included_images: list, enc_id: str
+):
+    """Compile Typst template into PDF"""
+    logger.info("Compiling PDF for %s → %s", enc_id, output_file)
 
+    with tempfile.TemporaryDirectory() as tmpdir:
+        template_path = Path(tmpdir) / "template.typ"
+        template_path.write_text(template_code)
 
-def generate_discharge_summary_pdf(data, file):
-    logger.info(
-        "Generating Discharge Summary pdf for %s", data["encounter"].external_id
-    )
-    compile_typ(output_file=file.name, data=data)
-    logger.info(
-        "Successfully Generated Discharge Summary pdf for %s",
-        data["encounter"].external_id,
-    )
+        for file_name, url in included_images:
+            logo_bytes = download_image_to_cache(file_name, url)
+            with Path.open(Path(tmpdir) / file_name, "wb") as f:
+                f.write(logo_bytes)
+
+        subprocess.run(  # noqa: S603
+            [settings.TYPST_BIN, "compile", str(template_path.name), str(output_file)],
+            cwd=tmpdir,
+            check=False,
```

Comment:
```suggestion
            check=True,
```
we should raise error in the case typst compile fails